patch-diff-report-tool/src/main/java/com/github/checkstyle/data/DiffReport.java

```diff
@@ -76,6 +77,7 @@ public Statistics getStatistics() {
     public void addRecords(List<CheckstyleRecord> newRecords,
             String filename) {
         if (!newRecords.isEmpty()) {
+            newRecords.sort(new PositionOrderComparator());
```

Comment:
Why we are using `PositionOrderComparator` for sorting lines within a file?
This comparator uses only `line` and `column`, as a result there may be false positives for
```
line 100, col 100, messageA
line 100, col 100, messageB
```
versus
```
line 100, col 100, messageB
line 100, col 100, messageA
```
I'm going to use same `specificCompare` method here

Reply:
We did not review this code thoroughly,  it is just a code that works

Reply:
Done. Well, now i wonder why `CheckstyleRecord` does not implement `Comparable` but instead exposes a method used in sorting? This should be reviewed.

Reply:
Just fix it if you think it weird

Reply:
done